src/Gaufrette/Adapter/Ftp.php

```diff
@@ -108,7 +108,7 @@ public function rename($sourceKey, $targetKey)
         $sourcePath = $this->computePath($sourceKey);
         $targetPath = $this->computePath($targetKey);
 
-        $this->ensureDirectoryExists(\Gaufrette\Util\Path::dirname($targetPath));
+        $this->ensureDirectoryExists(\Gaufrette\Util\Path::dirname($targetPath), $this->create);
```

Comment:
It would be better to enfore the creation of $targetPath. `$this->create` is used to ensure the base directory exists.